src/internet_identity/src/activity_stats/event_stats.rs

```diff
@@ -248,23 +251,54 @@ fn prune_events<M: Memory>(
 ) -> Vec<(EventKey, EventData)> {
     const RETENTION_PERIOD: u64 = 30 * DAY_NS;
 
-    let pruned_events = db
+    let pruned_events: Vec<_> = db
         .range(..=EventKey::max_key(now - RETENTION_PERIOD))
         .collect();
     for entry in &pruned_events {
         let entry: &(EventKey, EventData) = entry;
         db.remove(&entry.0);
     }
-
+    state::persistent_state_mut(|s| {
+        s.event_data_count -= pruned_events.len() as u64;
+    });
     pruned_events
 }
 
+/// Helper struct for updating the count of event aggregations.
+struct CountingAggregationsWrapper<'a, M: Memory>(&'a mut StableBTreeMap<AggregationKey, u64, M>);
```

Comment:
Why didn't you do something similar for the `event_data`?

Reply:
Because there is exactly one place where we insert (asserting the insert happened) and one bulk removal (where we do not need to modify the counter one by one). So it seemed more straight forward to just manipulate the counter in place.
But the logic gets any more complicated, then yes, we should add such a wrapper as well.

Reply:
Ok, makes sense. Thanks for the explanation!